### PR TITLE
[docs] Add an integration guideline page for In-App purchases

### PR DESCRIPTION
**Pull request by amandeepmittal**
# Why

Closes ENG-10452

Also, follows  #26422

# How

Provide a list of libraries and/or tutorials for integrating an In-app purchase library.

Also, add redirects to handle InAppPurchase API reference for the latest version (since we are near the release of SDK 50 to become the latest, it's a good practice to have redirects beforehand).

Also, add a redirect message when redirecting from SDK 50: https://github.com/expo/expo/pull/26425#issuecomment-1905784041

# Test Plan

Run docs locally and visit: http://localhost:3002/guides/in-app-purchases/ or see the [preview link](/guides/in-app-purchases/).

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: docs/pages/guides/in-app-purchases.mdx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  }
+  description="An open-source framework that provides a wrapper around Google Play Billing, StoreKit and RevenueCat. This library is a client for RevenueCat's subscription and purchase tracking system."
+  href="https://github.com/RevenueCat/react-native-purchases"
+  Icon={GithubIcon}
+/>
+
+<BoxLink
+  title={
+    <>
+      <CODE>react-native-iap</CODE>
```
Comment (amandeepmittal) [inline docs/pages/guides/in-app-purchases.mdx:45]: @keith-kurak, is it worth mentioning this library? In their [docs] or repo I do not see anything mentioned about Expo.

Also, are there any other libraries or resources we should mention here?

File: docs/pages/guides/in-app-purchases.mdx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+hideTOC: true
+---
+
+import { BoxLink } from '~/ui/components/BoxLink';
+import { CODE } from '~/ui/components/Text';
+import { BookOpen02Icon, GithubIcon } from '@expo/styleguide-icons';
+
+In-app purchases (IAP) are transactions within a mobile or desktop application where users can buy digital goods or additional features. This guide provides a list of popular libraries and tutorials for implementing IAP in your Expo app.
+
+> In-app purchase libraries require configuring custom native code. Native code is not configurable when using Expo Go. Instead, create a [development build](/develop/development-builds/introduction/), which will allow you to use a native library in your project.
```
Comment (amandeepmittal) [inline docs/pages/guides/in-app-purchases.mdx:13]:
```suggestion
> In-app purchase libraries require configuring custom native code. Native code is not configurable when using Expo Go. Instead, create a [development build](/develop/development-builds/introduction/), which allows using a native library in your project.
```

File: docs/pages/guides/in-app-purchases.mdx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+## Libraries
+
+<BoxLink
+  title={
+    <>
+      <CODE>react-native-purchases</CODE>
+    </>
+  }
+  description="An open-source framework that provides a wrapper around Google Play Billing, StoreKit and RevenueCat. This library is a client for RevenueCat's subscription and purchase tracking system."
```
Comment (keith-kurak) [inline docs/pages/guides/in-app-purchases.mdx:37]:
```suggestion
  description="An open-source framework that provides a wrapper around Google Play Billing and StoreKit API's, as well as integration with RevenueCat services supporting in-app purchases. `react-native-purchases` enables product management, analytics, and simplified workflows for in-app purchase requirements that may extend beyond your client code, such as validating purchases on your application backend."
```
maybe not the best wording, but wanted to emphasize the receipt validation angle, that's got a lot of appeal even if you don't like the other stuff.

File: docs/pages/guides/in-app-purchases.mdx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  Icon={GithubIcon}
+/>
+
+<BoxLink
+  title={
+    <>
+      <CODE>react-native-iap</CODE>
+    </>
+  }
+  description="A React Native library to access In-app purchases capabilities in your Android and iOS apps."
```
Comment (keith-kurak) [inline docs/pages/guides/in-app-purchases.mdx:48]:
```suggestion
  description="A React Native library providing an interface to the client-side native in-app purchase API's built into Android and iOS."
```
Trying to emphasize more that it's "just the wrapper".

File: docs/pages/guides/in-app-purchases.mdx [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+      The getting started guide for in-app purchases and subscriptions with{' '}
+      <CODE>react-native-purchases</CODE> library and RevenueCat.
+    </>
+  }
+  href="https://www.revenuecat.com/blog/engineering/expo-in-app-purchase-tutorial/"
+  Icon={BookOpen02Icon}
+/>
+
+## Libraries
+
```
Comment (keith-kurak) [inline docs/pages/guides/in-app-purchases.mdx:30]:
```suggestion

These libraries provide robust support for in-app purchase functionality, as well as out-of-the-box compatibility with Expo apps using [CNG](workflow/continuous-native-generation/). 
```
Some kind of intro as to why we're recommending these (they have config plugins + Expo-specific docs)

**Review by keith-kurak — COMMENTED** [review]
feedback added!

File: docs/pages/guides/in-app-purchases.mdx [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+These libraries provide robust support for in-app purchase functionality, as well as out-of-the-box compatibility with Expo apps using [CNG](workflow/continuous-native-generation/).
```
Comment (amandeepmittal) [inline docs/pages/guides/in-app-purchases.mdx:31]:
```suggestion
These libraries provide robust support for in-app purchase functionality and out-of-the-box compatibility with Expo apps using [CNG](workflow/continuous-native-generation/).

```

File: docs/pages/guides/in-app-purchases.mdx [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+<BoxLink
+  title={
+    <>
+      <CODE>react-native-purchases</CODE>
+    </>
+  }
+  description="An open-source framework that provides a wrapper around Google Play Billing and StoreKit API's, as well as integration with RevenueCat services supporting in-app purchases. `react-native-purchases` enables product management, analytics, and simplified workflows for in-app purchase requirements that may extend beyond your client code, such as validating purchases on your application backend."
+  href="https://github.com/RevenueCat/react-native-purchases"
+  Icon={GithubIcon}
+/>
```
Comment (amandeepmittal) [inline docs/pages/guides/in-app-purchases.mdx:41]:
```suggestion
  description={<><CODE>react-native-purchases</CODE> is an open-source framework that provides a wrapper around Google Play Billing and StoreKit APIs, as well as integration with RevenueCat services supporting in-app purchases. It enables product management, analytics, and simplified workflows for in-app purchase requirements that may extend beyond your client code, such as validating purchases on your app's backend.</>}
  href="https://github.com/RevenueCat/react-native-purchases"
  Icon={GithubIcon}
/>
```

File: docs/pages/guides/in-app-purchases.mdx [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+      <CODE>react-native-purchases</CODE> library and RevenueCat.
+    </>
+  }
+  href="https://www.revenuecat.com/blog/engineering/expo-in-app-purchase-tutorial/"
+  Icon={BookOpen02Icon}
+/>
+
+## Libraries
+
+These libraries provide robust support for in-app purchase functionality and out-of-the-box compatibility with Expo apps using [CNG](workflow/continuous-native-generation/).
```
Comment (amandeepmittal) [inline docs/pages/guides/in-app-purchases.mdx:31]:
```suggestion
The following libraries provide robust support for in-app purchase functionality and out-of-the-box compatibility with Expo apps using [CNG](workflow/continuous-native-generation/) and provide [Config Plugins](/config-plugins/introduction/) for seamless integration in your app.
```

**Comment by amandeepmittal** [comment]
@Simek, based on your feedback, I've made the following additions
- Renamed `VersionedRedirectNotification` to `RedirectNotification`
- Updated `RedirectNotification` references to include the redirect message in the `mdx` file
CleanShot 2024-01-23 at 16 16 32

- Added `RedirectNotification` to In App Purchase guide for SDK 50 and above
CleanShot 2024-01-23 at 16 14 12

**Review by keith-kurak — APPROVED** [review]

**Review by Simek — APPROVED** [review]
This looks great, thanks! 👍 

P.S. Let's also tinker next sync if there are any other situations in which we could benefit from using this redirect notice.

**Pull request merged**